ext/reflection/php_reflection.c

```diff
@@ -3106,19 +3151,116 @@ ZEND_METHOD(ReflectionNamedType, isBuiltin)
 }
 /* }}} */
 
-static void append_type(zval *return_value, zend_type type) {
+/* {{{ Resolve a relative class type to a proper named type */
+ZEND_METHOD(ReflectionRelativeClassType, resolveToNamedType)
+{
+	reflection_object *intern;
+	type_reference *param;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		RETURN_THROWS();
+	}
+	GET_REFLECTION_OBJECT_PTR(param);
+
+	/* Is closure */
+	if (!Z_ISUNDEF(intern->obj)) {
+		/* Unbound closures can use relative class types */
+		if (!intern->ce) {
+			ZEND_ASSERT(!Z_ISUNDEF(intern->obj));
+			zend_throw_exception_ex(reflection_exception_ptr, 0,
+				"Cannot resolve relative class name for a closure");
```

Comment:
Suggestion: "Cannot resolve relative class name for a static closure" (because we can resolve for some closures)